### PR TITLE
Add e2e test for hosted cluster behind a proxy

### PR DESCRIPTION
**Pull request by enxebre**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Ref https://issues.redhat.com/browse/HOSTEDCP-731
blocked by https://issues.redhat.com/browse/OCPBUGS-6651

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by enxebre** [comment]
Needs https://github.com/openshift/cluster-network-operator/pull/1694

**Comment by enxebre** [comment]
needs https://github.com/openshift/cluster-storage-operator/pull/336
/test e2e-aws

File: test/e2e/create_cluster_test.go [hunk 1]
```diff
@@ -143,3 +143,57 @@ func TestCreateClusterPrivate(t *testing.T) {
...
+	ctx, cancel := context.WithCancel(testContext)
+	defer cancel()
+
+	client, err := e2eutil.GetClient()
+	g.Expect(err).NotTo(HaveOccurred(), "failed to get k8s client")
+
+	clusterOpts := globalOpts.DefaultClusterOptions(t)
+	clusterOpts.AWSPlatform.EnableProxy = true
+	zones := strings.Split(globalOpts.configurableClusterOptions.Zone.String(), ",")
+	if len(zones) >= 3 {
```
Comment (sjenning) [inline test/e2e/create_cluster_test.go:161]: We don't need this one more than one test.  CreateCluster has it covered.

**Review by sjenning — CHANGES_REQUESTED** [review]

**Pull request merged**